### PR TITLE
Add installIn parameter to SoloModule

### PR DESCRIPTION
**Pull request by KVelasco**
## Breaking Change 

Before this change Pommel would automatically install the generated `SoloModule` into the correct component based on the scope you gave your dependency. 

use of `Pommel 0.1.0 `

```kotlin
@SoloModule
@Singleton
class SampleClass @Inject constructor(
    @Named("a") val a: Int,
    val b: String
)
```

would generate the equivalent of:

```java
@Module
@InstallIn(SingletonComponent.class)
public abstract class SampleClass_SoloModule {
  @Provides
  @Singleton
  public static SampleClass provides_SampleClass(@Named("a") int a, String b) {
    return new SampleClass(a, b);
  }
}
```

This had a few drawbacks the biggest of which was the inability to use a custom component with Pommel. 

To use a custom component before this change you would have to set `install` to false and then manually include the generated module into the component which would be  quite cumbersome to deal with if you had a lot of custom components that you were using in your application.

Another drawback is that pommel would always install in the `SingletonComponent` when having an unscoped dependency. It might not always be the case that you want to install a unscope dependency into  in the `SingletonComponent`. There are times when you might want to have an unscope dependency in the `ActivityComponent`  or have scope dependency that does not directly map to a component such as `@Reusable` 

use of `Pommel 0.1.0 ` with custom scope 

This would not compile as `@Reusable`. does not map to a component 
```kotlin
// fails to compile due to use of @Resuable 
@SoloModule
@Reusable 
class SampleClass @Inject constructor(
    @Named("a") val a: Int,
    val b: String
)
```

use of `Pommel 0.1.0 ` with no scope 

```kotlin
@SoloModule
class SampleClass @Inject constructor(
    @Named("a") val a: Int,
    val b: String
)
```

Would always install in SingletonComponent:

```java
@Module
@InstallIn(SingletonComponent.class) // You might not always want this 
public abstract class SampleClass_SoloModule {
  @Provides
  public static SampleClass provides_SampleClass(@Named("a") int a, String b) {
    return new SampleClass(a, b);
  }
}
```

Since Pommel is designed to be integrated with Hilt it now forces the developer to declared a component when annotating a `class` or `function` with `@SoloModule`. 

So now using pommel would look something like this: 

```kotlin
@SoloModule(installIn = SingletonComponent::class)
class SampleClass @Inject constructor(
    @Named("a") val a: Int,
    val b: String
)
```

Will generate the equivalent of:

```java
@Module
@InstallIn(SingletonComponent.class)
public abstract class SampleClass_SoloModule {
  @Provides
  public static SampleClass provides_SampleClass(@Named("a") int a, String b) {
    return new SampleClass(a, b);
  }
}
```

If there ever becomes a need to use Pommel with vanilla Dagger or Dagger-Android please file a Github issue with your usecase

### CODE REVIEW

**Comment by twyatt** [comment]
Copy/paste w/o updating in PR description? 😄 

Screen Shot 2020-08-27 at 8 15 56 PM

File: README.md [hunk 1]
```diff
@@ -118,7 +95,7 @@ If your class extends an `abstract class` or `interface`, you can specify to bin
 ```kotlin
 interface MyInterface
 
-@SoloModule(MyInterface::class)
+@SoloModule(MyInterface::class, SingletonComponent::class)
```
Comment (twyatt) [inline README.md:98]: Having these named would make it more clear what the parameters are for in the documentation.

File: README.md [hunk 2]
```diff
@@ -143,7 +120,7 @@ You can annotate your class with a qualifier:
...
```
Comment (twyatt) [inline README.md:123]: Same here.

File: README.md [hunk 3]
```diff
@@ -330,13 +309,11 @@ Reference the [Dagger-Hilt testing package](https://dagger.dev/hilt/testing) for
...
+Pommel expects all functions annotated with `@SoloModule` to be `static` due to the fact that the generated module needs to call the annotated function.
+As a result Pommel only supports top level functions and functions that are enclosed in an `object` or `companion` class
 
-Pommel will install your dependency into the correct component given the scope your dependency was annotated with. If you do not provide a scope to the dependency Pommel will by default install into the SingletonComponent.
-If you annotate your dependency with a custom scope you will get a compiler error as Pommel does currently not support being installed into a custom component.
-You can get around this current limitation by setting the `install` parameter to false and pommel will skip generating the `InstallIn` annotation on your module.
-The downside is you now have to manually include the generated module into your dagger graph rather the module being auto installed in for you.
+Pommel was designed to be used with Dagger-Hilt as result it expects a component to be passed into the `installIn` parameter. The component passed into `installIn`  follows the same limitation of Dagger-Hilt.
```
Comment (twyatt) [inline README.md:315]:
```suggestion
Pommel was designed to be used with Dagger-Hilt so it expects a component to be passed into the `installIn` parameter. The component passed into `installIn`  follows the same limitations imposed by Dagger-Hilt.
```

**Review by twyatt — APPROVED** [review]

**Comment by KVelasco** [comment]
> Copy/paste w/o updating in PR description? 😄
@twyatt Whoops! That was a copy and paster error. But you get the gist lol

**Review by twyatt — APPROVED** [review]

**Review by Chris-Corea — APPROVED** [review]

**Pull request merged**